control-plane/grpc/proto/v1/nexus/nexus.proto

```diff
@@ -106,6 +106,8 @@ message NexusSpec {
   optional common.SpecOperation operation = 10;
   // Nexus Nvmf Configuration
   optional NexusNvmfConfig nvmf_config = 11;
+  // Hosts allowed to access nexus
+  repeated string allowed_hosts = 16;
```

Comment:
12?

Reply:
Should this be part of the nvmf_config?

Reply:
nvmf_config is used while forming CreateNexus struct. This is for ShareNexus param so i think it should be outside nvmf_config.

Reply:
Whilst it's used in the create, it doesn't have to be limited to that. We can simply get the allowed_hosts from the nexus config when sharing if needed. It's better to keep it compacted IMO otherwise we'll making the already quite large spec ever larger.

Reply:
Im thinking to make spec compact we can group Sharenexus parameters which are outside currently. Maybe there can be another PR to restructure all struct if applicable.  For CreateNexus.config we directly pass Option<NexusNvmfConfig>. It seems wrong somehow to pass an unncessary value to that struct.